### PR TITLE
Fix pcsaft flash

### PR DESCRIPTION
**Pull request by zmeri**
### Requirements

* Fill out this template to the extent possible so that this PR can be reviewed in a timely manner.
* Replace the bracketed text below with your own.
* All new code requires tests to ensure against regressions.

### Description of the Change

The flash and density solvers for the PC-SAFT backend were update to make them more reliable. Some minor edits to function names were also added to make it clear that PC-SAFT calculates residual H, S, and G.

### Benefits

Improved consistency and higher likelihood that PC-SAFT backend will find the correct solution.

### Possible Drawbacks

That code elsewhere might break

### Verification Process

I added additional unit tests to confirm that the flash and density functions worked over a wider range of conditions, and all these unit tests passed. I also added a python script to make consistency plots for the PC-SAFT backend, and the consistency improved significantly after the changes. I also ran all the unit tests and almost all passed. Of 50497 assertions only 240 failed. Most of those that failed were because I do not have the Refprop or UNIFAC libraries, and the few others I checked also failed before the changes made here, so they appear to be unrelated to the changes here.

### Applicable Issues

[ *Enter any applicable Issues here.  Use ``Closes #????`` if this PR closes an open issue.* ]

### CODE REVIEW

File: include/all_pcsaft_JSON.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// File generated by the script dev/generate_headers.py on 2019-12-12 21:03:32.480501
+// File generated by the script dev/generate_headers.py on 2020-04-07 19:35:54.752805
```
[the remark this answers is not in the document]
Comment (zmeri) [reply include/all_pcsaft_JSON.h:2]: I added this to the .gitignore file.

File: src/pybind11_interface.cxx [hunk 2]
```diff
@@ -110,6 +110,8 @@ void init_CoolProp(py::module &m){
         .value("iGmolar", parameters::iGmolar)
         .value("iHelmholtzmolar", parameters::iHelmholtzmolar)
         .value("iSmolar_residual", parameters::iSmolar_residual)
+        .value("iHmolar_residual", parameters::iHmolar_residual)
+        .value("iGmolar_residual", parameters::iGmolar_residual)
         .value("iDmass", parameters::iDmass)
```
[the remark this answers is not in the document]
Comment (zmeri) [reply src/pybind11_interface.cxx:115]: I updated the AbstractState pyx and pxd files. Are there any other files I need to update?

File: include/all_pcsaft_JSON.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibell) [reply include/all_pcsaft_JSON.h:2]: Can you also remove the generated JSON files from source control?  They aren't enormous, but every byte counts.

File: wrappers/Python/CoolProp/AbstractState.pxd [hunk 3]
```diff
@@ -156,10 +156,13 @@ cdef class AbstractState:
     cpdef double helmholtzmolar_excess(self) except *
     cpdef double helmholtzmass_excess(self) except *
 
-    
+    cpdef double gibbsmolar_residual(self) except *
+    cpdef double hmolar_residual(self) except *
+    cpdef double smolar_residual(self) except *
```
Comment (ibell) [inline wrappers/Python/CoolProp/AbstractState.pxd:161]: I think they also need to get added to cAbstractState.pxd . This code duplication for C++ classes is one reason that I much prefer pybind11 for wrapping C++.  Perhaps you can test compilation for Python?

File: include/all_pcsaft_JSON.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (zmeri) [reply include/all_pcsaft_JSON.h:2]: Ah yes, I forgot to do that. Done.

File: wrappers/Python/CoolProp/AbstractState.pxd (hunk 3 above)
Comment (zmeri) [reply wrappers/Python/CoolProp/AbstractState.pxd:161]: I added them to cAbstractState.pxd as well. I tested compilation and ran a small Python script to check that it worked. The PC-SAFT values matched those from the HEOS backend.

**Comment by ibell** [comment]
Thanks, merged.

**Pull request merged**